src/usr/share/netcheck/networkmanagerhelper.py

```diff
@@ -413,6 +413,35 @@ def connection_is_activated(self, connection_id):
 
         return connection_is_activated
 
+    @reiterative
+    def get_connection_for_interface(self, interface_name):
+        """Find the connection ID currently applied to the given interface.
+
+        interface_name: The system interface name, for example eth1 or ens0s1.
+
+        Returns the connection ID.
+        """
+        connection_id = None
+        devices = []
+
+        try:
+            devices = self.NetworkManager.NetworkManager.GetDevices()
+
+            for device in devices:
+                if device.Interface == interface_name:
+                    connection_settings = self._get_applied_connection(device)
+                    if connection_settings:
+                        connection_id = connection_settings['connection']['id']
+
+        # TODO: Implement exception chaining when we move to Python 3.
+        except Exception as exception:
+            self.logger.error(
+                'Unable to get device list from NetworkManager. %s: %s.',
```

Comment:
This line is not accurate if connection_settings is None. Can you make this more general?